crates/starknet/src/starknet/mod.rs

```diff
@@ -783,6 +787,32 @@ impl Starknet {
 
         Ok(simulation_results)
     }
+
+    pub fn create_block(&mut self) -> DevnetResult<(), Error> {
+        // create new block from pending one
+        self.generate_new_block(StateDiff::default())?;
```

Comment:
In the case of still not developed feature "Block on demand/Pending block" this will lead to some unexpected behavior when extracting data from state.

Reply:
So there are 2 paths to take, either implement similar logic as handle_successful_transaction or refactor it

Reply:
is it still relevant after changes in 58460a7974dfef8f37839b83789402e124b682d4 and the merge of  https://github.com/0xSpaceShard/starknet-devnet-rs/pull/235 PR?

Reply:
For me this conversation is resolved, leaving to Fabijan (@FabijanC) for any additional comments, if any

Reply:
I was just worried because this `StateDiff::default()` is not used **only** in tests now. But if you think it's good like this, I'll resolve it.